IPAM for docker interfaces

jorisc90 opened the pull request:
Added IPAM for docker-topo to be able to have their IP subnet the same as OOB_PREFIX. This makes it possible to reach containers using HTTPS using macvlan connections to the outside without extra configuration within the container.

Commit:
added IPAM in the network creation, such that networks that get created by docker-topo have their IP subnet the same as OOB_PREFIX. This makes it possible to reach containers using HTTPS using macvlan connections to the outside without extra configuration within the container.

Commit:
change to env python3

networkop commented:
looks good, just need to test it. will do it lqter this week

Commit:
added the IPPREFIX and GATEWAY IPAM options to both the script and the documentation.

Commit:
fix readme typo

Commit:
fix readme typo

jorisc90 commented:
Added extra changes which allow it to have more flexibility when configuring these networks. No impact to the current script.

Commit:
fix the python3 env

networkop commented:
I think your patch is introducing a bug. If you have more than just one "bridge" driver network, the script will crash with the following reason:
docker.errors.APIError: 500 Server Error: Internal Server Error ("failed to allocate gateway (192.168.100.1): Address already in use")

This is because you can't allocate the same IPAM settings to multiple networks.

Can you help me understand what your goal is?

networkop committed:
fix example name

networkop committed:
Merge branch 'master' of github.com:networkop/arista-ceos-topo

networkop committed:
add ssh exposed ports

networkop committed:
publish any ports

networkop committed:
publish any port

networkop committed:
publish any port

networkop committed:
remove old example

alexisdacquay committed:
Create 5-node-ls.yml

Create a v2 config example for 5-nodes Leaf-Spine topology

alexisdacquay committed:
Creates a topology of 14 devices

alexisdacquay committed:
Removes the comments as they seem unsupported by the parsing in the topo script.

alexisdacquay committed:
Reduces the PREFIX length to something short, as it was causing issues.

alexisdacquay committed:
Reduces the PREFIX name length

alexisdacquay committed:
Diagram to illustrate the 14rtr topology

alexisdacquay committed:
Corrects missing endpoints in the 14rtr.yml file and incorrect link mapping on rtr13.
Adds 14 individual configs in a directory
Adds the original topology diagram (draw.io version .html)

alexisdacquay committed:
Change the filename from a ".log" extension to not any extension.

alexisdacquay committed:
Updates the PNG version of the topology diagram

alexisdacquay committed:
Updates the topology diagrams of the 14rtr example

networkop committed:
Merge pull request #11 from alexisdacquay/master

Looks good. Thanks Alexis

Commit:
update for cEOS 4.21.0F

Commit:
Merge branch 'master' of github.com:jorisc90/arista-ceos-topo
